.circleci/config.yml

```diff
@@ -1522,6 +1522,13 @@ jobs:
   c_ext_benchmarks:
     <<: *base_node_small
     steps:
+      - run:
+          name: Check CircleCI token presence; Skip job if not set.
+          command: |
+            if [[ -z "$CIRCLECI_TOKEN" ]]; then
+              echo "Skipping benchmarks..."
+              circleci-agent step halt
+            fi
```

Comment:
If it's here, it skips the whole job, but we could easily keep the first part, which is just combining the reports and attaching them as an artifact. This is still useful, because it allows the contributor to at least download files manually and diff them locally. The only bit that needs to be skipped is benchmark download from base branch and diffing.

Reply:
right, but then it would be kind of incorrect. I mean, we would need to rename that variable because it would not be skipping only the benchmark diffs. I guess it is good enough in the way that it is now, no?

Reply:
Not really. Currently we do something like this (pseudo code):

```
if !download_benchmarks.py:
    SKIP_BENCHMARK_DIFF=true
```

And this is what I meant:

```
if !has token || !download_benchmarks.py:
    SKIP_BENCHMARK_DIFF=true
```

So this does not really change the meaning of the variable. It still only skips diffing. It's the token check that skips the download.

Reply:
But yeah, the current way is not wrong either. Just this seemed much simpler to me.